File: seatmaps/v1/post/Python SDK/seatmaps.py
# Install the Python library from https://pypi.org/project/amadeus
from amadeus import Client, ResponseError

amadeus = Client(
    client_id='YOUR_AMADEUS_API_KEY',
    client_secret='YOUR_AMADEUS_API_SECRET'
)

try:
    '''
    Retrieve the seat map of a given flight offer 
    '''
    body = amadeus.shopping.flight_offers_search.get(originLocationCode='MAD',
                                                        destinationLocationCode='NYC',
                                                        departureDate='2020-11-01',
                                                        adults=1,
                                                        max=1).result
    response = amadeus.shopping.seatmaps.post(body)
    print(response.data)
except ResponseError as error:
    raise error

File: seatmaps/v1/post/Node SDK/seatmaps.js
const Amadeus = require("amadeus");

const amadeus = new Amadeus({
  clientId: "YOUR_API_KEY",
  clientSecret: "YOUR_API_SECRET",
});

async function main() {
  try {
    // Returns all the seat maps of a given flightOffer
    const flightOffersResponse = await amadeus.shopping.flightOffersSearch.get({
      originLocationCode: "SYD",
      destinationLocationCode: "BKK",
      departureDate: "2022-08-01",
      adults: "2",
    });

    const response = await amadeus.shopping.seatmaps.post({
      data: [flightOffersResponse.data[0]],
    });

    console.log(response);
  } catch (error) {
    console.error(error);
  }
}

main();
